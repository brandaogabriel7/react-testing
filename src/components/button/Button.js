import React from 'react';
import './button.css';

const Button = ({ label }) => {
    return <button className="button-style">{label}</button>;
};

export default Button;