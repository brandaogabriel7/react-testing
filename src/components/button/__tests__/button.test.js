import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { render, screen } from '@testing-library/react';

import Button from '../Button';

describe('button should', () => {
    it('render correctly', () => {
        render(<Button label="click me please" />);
        expect(screen.getByRole('button', { name: /click me please/i })).toBeInTheDocument();
    });

    it('render correctly', () => {
        render(<Button label="whaaat" />);
        expect(screen.getByRole('button', { name: /whaaat/i })).toBeInTheDocument();
    });

    it('match snapshot 1', () => {
        render(<Button label="save" />);
        expect(screen.getByRole('button', { name: /save/i })).toMatchSnapshot();
    });

    it('match snapshot 2', () => {
        render(<Button label="submit" />);
        expect(screen.getByRole('button', { name: /submit/i })).toMatchSnapshot();
    });
});