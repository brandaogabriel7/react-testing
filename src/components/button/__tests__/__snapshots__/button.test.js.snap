// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button should match snapshot 1 1`] = `
<button
  class="button-style"
>
  save
</button>
`;

exports[`button should match snapshot 2 1`] = `
<button
  class="button-style"
>
  submit
</button>
`;
